### PR TITLE
Long Mode

### PR DESCRIPTION
**Pull request by Restioson**
This PR allows the kernel to transition to long mode. That's all, basically.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -4,4 +4,3 @@ src/rust/Cargo.lock
 .idea/
 build/
 flower.iml
-isofiles/
```
Comment (Restioson) [inline .gitignore]: Unrelated change (unfortunately)

File: src/boot/boot.asm [hunk 2]
```diff
@@ -12,35 +12,26 @@ bits 32
...
-; Formula: memory loc = VGA_PTR + (y * columns + x) * 2
-;          => L = VGA_PTR + (y * 80 + x) * 2
-; *note, the times 2 is because vga buffer is 16bit
-; Columns = 80 because the res is 80x25
-clear_screen:
-    mov ecx, (RESOLUTION_Y * RESOLUTION_X + RESOLUTION_X) * 2 ; bottom right pixel (without video buffer pointer offset)
+    ; Checks
+    ;call check_multiboot ; Check if booted correctly
+    ;call check_cpuid  ; Check if cpuid supported
+    ;call check_long_mode ; Check if long mode supported
```
Comment (gegy1000) [inline src/boot/boot.asm]: Should be uncommented.

File: src/boot/boot.asm [hunk 3]
```diff
@@ -129,10 +321,10 @@ boot_print:
...
     mov word [VGA_PTR + 14], 0x0253 ; S
-    mov word [VGA_PTR + 16], 0x0220 ;
+    mov word [VGA_PTR + 16], 0x0220 ;  
     mov word [VGA_PTR + 18], 0x0262 ; b
     mov word [VGA_PTR + 20], 0x026f ; o
     mov word [VGA_PTR + 22], 0x026f ; o
     mov word [VGA_PTR + 24], 0x0274 ; t
     
-    ret
+    ret
```
Comment (gegy1000) [inline src/boot/boot.asm]: Should have newline at end of file.

File: src/boot/multiboot_header.asm [hunk 4]
```diff
@@ -1,19 +1,16 @@
-; Boot headers for multiboot 2
-; This lets the OS boot with any multiboot 2 compatible bootloader
-; GRUB is the chosen bootloader for its ubiquity
```
Comment (gegy1000) [inline src/boot/multiboot_header.asm]: Redundant newline here.

**Comment by Restioson** [comment]
Accidentally PR'd into master, not development
